src/Paket.Core/Utils.fs

```diff
@@ -196,6 +196,12 @@ let inline createWebClient(url,auth:Auth option) =
     client.Proxy <- getDefaultProxyFor url
     client
 
+//More generic variant of http://www.fssnip.net/c4
+open System.Collections.Concurrent
+let cache = ConcurrentDictionary<(string * obj) option,Lazy<obj>>()
+let memoizeConcurrent (caller:string) (f: ('a -> 'b)) =
+    fun (x :'a) -> 
+        (cache.GetOrAdd(Some (caller, x|>box), lazy ((f x)|>box)).Force() |> unbox) : 'b
```

Comment:
why the `lazy ((f x) |> box)).Force()`?
why not `(f x) |> box)`?

Why is the key `(string * obj) option` and not just `(string * obj)`?